src/ripple/app/paths/Pathfinder.cpp

```diff
@@ -514,47 +530,58 @@ void Pathfinder::rankPaths (
     STPathSet const& paths,
     std::vector <PathRank>& rankedPaths)
 {
-
     rankedPaths.clear ();
     rankedPaths.reserve (paths.size());
 
-    // Ignore paths that move only very small amounts.
-    auto saMinDstAmount = smallestUsefulAmount (mDstAmount, maxPaths);
+    STAmount saMinDstAmount;
+    if (convert_all_)
+    {
+        saMinDstAmount = STAmount(mDstAmount.issue(),
+            STAmount::cMaxValue, STAmount::cMaxOffset);
```

Comment:
This is pretty obscure. This looks like the code will (likely) never find any paths because it requires _all the monies_. It's only later in `getPathLiquidity` that `partialPaymentAllowed` gets set to true. I think a comment here explaining the situation would be sufficient.